nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ExtractGrok.java

```diff
@@ -107,31 +120,70 @@
         .build();
 
     public static final PropertyDescriptor CHARACTER_SET = new PropertyDescriptor.Builder()
-        .name("Character Set")
+        .name(CHARACTER_SET_KEY)
         .description("The Character Set in which the file is encoded")
         .required(true)
         .addValidator(StandardValidators.CHARACTER_SET_VALIDATOR)
         .defaultValue("UTF-8")
         .build();
 
     public static final PropertyDescriptor MAX_BUFFER_SIZE = new PropertyDescriptor.Builder()
-        .name("Maximum Buffer Size")
+        .name(MAXIMUM_BUFFER_SIZE_KEY)
         .description("Specifies the maximum amount of data to buffer (per file) in order to apply the Grok expressions. Files larger than the specified maximum will not be fully evaluated.")
         .required(true)
         .addValidator(StandardValidators.DATA_SIZE_VALIDATOR)
         .addValidator(StandardValidators.createDataSizeBoundsValidator(0, Integer.MAX_VALUE))
         .defaultValue("1 MB")
         .build();
 
-    public static final PropertyDescriptor NAMED_CAPTURES_ONLY = new PropertyDescriptor.Builder()
-        .name("Named captures only")
-        .description("Only store named captures from grok")
+     public static final PropertyDescriptor NAMED_CAPTURES_ONLY = new PropertyDescriptor.Builder()
+        .name(NAMED_CAPTURES_ONLY_KEY)
+        .description("Only store named captures from grokList")
         .required(true)
         .allowableValues("true", "false")
         .addValidator(StandardValidators.BOOLEAN_VALIDATOR)
         .defaultValue("false")
         .build();
 
+    public static final PropertyDescriptor BREAK_ON_FIRST_MATCH = new PropertyDescriptor.Builder()
+        .name(SINGLE_MATCH_KEY)
+        .description("Stop on first matched expression.")
```

Comment:
Can you explain better what this property is meant to do? How does the behavior of the processor change if i select 'true' vs. if I select 'false'?

Reply:
glad to get early warning that it was not intelligible.  I will add a more complete just FYI: in my use case. i am using this processor as a classifier. I arrange my expressions, so  that the first match it makes will be the best match..